src/Sylius/Bundle/OrderBundle/NumberGenerator/OrderNumberGenerator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is a part of the Lakion package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\OrderBundle\NumberGenerator;
+
+use Doctrine\ORM\EntityManager;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Order\Model\OrderSequenceInterface;
+use Sylius\Component\Resource\Factory\Factory;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class OrderNumberGenerator implements OrderNumberGeneratorInterface
```

Comment:
Anyway, this class is not only a generator, it is an assigner also. I would suggest to split this logic, and create a generator, which will only generate number and increment a sequence and a second one, which will check if a number generation is require and if so, just call a generate method. Checker could be also extracted, but I'm afraid that it would be an overengineering.